internal/integration/initramfs_test.go

```diff
@@ -0,0 +1,203 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+//go:build integration
+
+package integration_test
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"slices"
+	"strings"
+	"testing"
+
+	"github.com/siderolabs/gen/optional"
+	"github.com/siderolabs/gen/xslices"
+	"github.com/siderolabs/talos/pkg/machinery/extensions"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/u-root/u-root/pkg/cpio"
+	"github.com/ulikunitz/xz"
+	"gopkg.in/yaml.v3"
+)
+
+type initramfsSpec struct {
+	earlyPaths      []string
+	extensions      []string
+	modulesDepMatch optional.Optional[string]
+	flavorID        string
+}
+
+func eatPadding(t *testing.T, in *bufio.Reader) {
+	t.Helper()
+
+	for {
+		b, err := in.ReadByte()
+		if err == io.EOF {
+			return
+		}
+
+		require.NoError(t, err)
+
+		if b != 0 {
+			require.NoError(t, in.UnreadByte())
+
+			return
+		}
+	}
+}
+
+func assertInitramfs(t *testing.T, path string, expected initramfsSpec) {
+	t.Helper()
+
+	f, err := os.Open(path)
```

Comment:
not closed?

Reply:
it's closed at the very bottom of this function